generic/image.c

```diff
@@ -15,12 +15,19 @@
 #define M_PI    3.14159265358979323846
 #endif
 
+#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE)
```

Comment:
`#undef temp_t` should be inserted before to avoid `'temp_t' macro redefined` warnings.